### PR TITLE
Base64 and consistent URL encoding

### PR DESCRIPTION
**Pull request by paulstraw**
Hey @parkr, this is my first ever Go pull request, so please let me know if anything here could be more idiomatic. 😁

### CODE REVIEW

File: imgix.go [hunk 1]
```diff
@@ -153,11 +154,22 @@ func (c *Client) PathWithParams(imgPath string, params url.Values) string {
...
+			params.Set(key, base64EncodedVal)
+		}
+	}
+
 	// The signature in an imgix URL must always be the **last** parameter in a URL,
 	// hence some of the gross string concatenation here. net/url will aggressively
 	// alphabetize the URL parameters.
 	signature := c.SignatureForPathAndParams(imgPath, params)
 	parameterString := params.Encode()
+	parameterString = strings.Replace(parameterString, "+", "%%20", -1)
```
Comment (parkr) [inline imgix.go]: Could you document why this double percent sign is necessary and perhaps shed some light on why a + isn't valid?

File: imgix.go [hunk 2]
```diff
@@ -153,11 +154,22 @@ func (c *Client) PathWithParams(imgPath string, params url.Values) string {
...
 	urlString += imgPath
 
+	for key, val := range params {
+		if strings.HasSuffix(key, "64") {
+			valData := []byte(val[0])
+			base64EncodedVal := base64.URLEncoding.EncodeToString(valData)
+			base64EncodedVal = strings.Replace(base64EncodedVal, "=", "", -1)
+			params.Set(key, base64EncodedVal)
+		}
+	}
```
Comment (parkr) [inline imgix.go]: Could you place this in its own unexported method and provide documentation for why it's necessary?

**Comment by parkr** [comment]
Hey, @paulstraw! :wave: Thanks for the PR. I have two questions above for you. Let me know if you have any questions! :smile:

File: imgix.go (hunk 1 above)
Comment (paulstraw) [inline imgix.go]: I may be misunderstanding Go string escaping, but it seemed that the `%%` was required in order to properly output the final `%20` string in the output. Is this not the case? The tests show that this works, but again, I might totally be missing something.

As for the rationale, either are technically valid. In our particular case, encoding space characters as `%20` helps keeps output consistent among all our libraries.

File: imgix.go (hunk 2 above)
Comment (paulstraw) [inline imgix.go]: Sure thing, I can move it.

Our new Base64 variant params are designed to make advanced usage of imgix easier. They work especially well with `blend`, `mark`, and `txt` params that are nested multiple times.

**Comment by paulstraw** [comment]
Added a couple notes. I'll get the B64 stuff pulled out to its own method. Let me know if you have any other questions/comments.

File: imgix.go (hunk 1 above)
Comment (parkr) [inline imgix.go]: > it seemed that the %% was required in order to properly output the final %20 string in the output.

You're absolutely correct. `%` is used to designate interpolation, and a literal `%` must be designated as `%%`. 👍

**Comment by paulstraw** [comment]
Should be set now I believe, @parkr. Let me know if there are any other questions/issues.

**Comment by parkr** [comment]
Excellent, thank you!

**Pull request merged**